client/state/signup/steps/site-style/selectors.js

```diff
@@ -0,0 +1,11 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import { get } from 'lodash';
+
+export function getSiteStyle( state ) {
+	return get( state, 'signup.steps.siteStyle', '' );
```

Comment:
I mentioned this elsewhere, but what do you think about storing this data in a key called `dependencies` rather than steps? My reasoning is that I think it makes sense to separate the dependencies from the steps, as the same dependency can be provided by steps of different names in different flows, so it think it makes sense to separate them.

Reply:
> I mentioned this elsewhere, but what do you think about storing this data in a key called dependencies rather than steps? ... as the same dependency can be provided by steps of different names in different flows

Thanks for looking at this. Your reasoning makes sense to me. Would we change the key for all related reducers as well? 

One question, what's the relationship between the key/value pairs we're storing in `state.signup.dependencyStore` and the step values? Could we not just use those?

Reply:
@scruffian If I'm not mistaken, this is actually what eventually will happen in the up-coming PRs.

As @ramonjd pointed out, we do have a separate state for dependencies, `state.signup.dependencyStore`. It's a redux substate tree filled by the [Flux-redux compatible layer](https://github.com/Automattic/wp-calypso/blob/master/client/lib/signup/dependency-store.js#L53), through `SignupActions.submitStep()`. Thus, the `steps.siteStyle` state here actually does what you expected: it's a step-specific app state which is separated from signup dependencies and likely happens to have the same value.

Does that make sense to you?

> Could we not just use those?

In my opinion, it's tempting but risky. I'm all for de-duping any data whenever possible. However, at least for now the two still stand for a bit different purpose: one for necessary signup data, and one for the app state representing a specific step. I'd suggest we keep them both for now.

Reply:
My issue with the with current structure is that it assumes as 1-1 relationship between steps and dependencies, which should not be the case. Dependencies can be provided by steps of different names, or even in the query string. We need to ensure that the framework we are building is able to maintain this separation between steps and dependencies.

Reply:
> Is there somewhere in the code to illustrate this?

Not sure if this illustrates what @scruffian had in mind, but here's one place in the code where the separations become fuzzy:
 When a `?vertical=restaurant` query param is present, we fill the dependency state and then exclude the site topic step. We shouldn't have to set the steps state, because the user never goes through that step, but we do end up filling the step state [here](https://github.com/Automattic/wp-calypso/blob/master/client/signup/main.jsx#L283) because as you rightly pointed @ramonjd , we are not using `SignupDependencyStore.get()` when calling the _/sites_new_ endpoint.

> My instinct is to reduxify SignupDependencyStore first.

I believe this was attempted before but then kept on the backburner due to some issues that cropped up. From what I understand, reduxifying - though has its benefits - was not [initially planned](https://github.com/Automattic/wp-calypso/pull/28209#discussion_r230268994) as a priority task for phase 1. @michaeldcain would be better equipped to comment on that though.

Reply:
One more thing to add: the `_canProcessStep()` in `flow-controller` is using the step progress store to check if all the necessary steps are completed: https://github.com/Automattic/wp-calypso/blob/master/client/lib/signup/flow-controller.js#L223 , and that's one more factor that we can't just fill in the dependency store and get the `apiRequestFunction` fired.